core/executor/src/native_executor.rs

```diff
@@ -59,28 +62,78 @@ pub trait NativeExecutionDispatch: Send + Sync {
 	fn native_version() -> NativeVersion;
 }
 
+/// The callable Substrate runtime. When called, a version of the Wasm runtime code is executed.
+pub trait SubstrateRuntime {
```

Comment:
While I also don't find `SubstrateRuntime` a good term, I think `WasmRuntime` might be not much better. This is because wasm runtime can mean something that executes wasm. OTOH in the context of substrate it means the wasm version of substrate runtime.